src/core/core.lisp

```diff
@@ -123,18 +133,32 @@
   (print-unreadable-object (object stream :type t)
     (format stream "addr=~A" (mac-address object))))
 
-(defmethod mac-octets ((object magic-packet))
-  (make-array 6 :element-type '(unsigned-byte 8) :initial-contents (parse-hex-bytes (mac-address object))))
-
-(defun make-magic-packet (mac-address &optional password)
-  "Creates a new instance of MAGIC-PACKET with the given MAC-ADDRESS"
-  ;; Validate MAC addr
-  (unless (cl-ppcre:scan *mac-regex* (string-downcase mac-address))
-    (error 'invalid-mac-address :mac-address mac-address))
-  ;; Validate SecureOn password
-  (when (and password (not (parse-hex-bytes password)))
-    (error 'invalid-password :password password))
-  (make-instance 'magic-packet :mac-address mac-address :password password))
+(defmethod mac-address ((object magic-packet))
+  (with-output-to-string (s)
+    (loop :for (byte . rest) :on (map 'list #'identity (mac-octets object)) :do
+      (if rest
+	  (format s "~2,'0x:" byte)
+	  (format s "~2,'0x" byte)))))
```

Comment:
Tabs and spaces are mixed in indentation of this file - I'd suggest to untabify and reindent everything because of that.

Reply:
Sigh. My `before-save-hook` apparently got lost after a recent switch from SLIME to Sly. Fixed in latest commit.